proto/substrait/algebra.proto

```diff
@@ -561,14 +561,63 @@ message WriteRel {
   }
 }
 
+// Equality joins (equi-joins) are a specialization of the general join where the join expression
+// is an series of equality comparisons between fields that are ANDed together.  The exact definition
+// of "equality" is flexible.
+message EquiJoinKey {
+  // The key to compare from the left table
+  Expression.FieldReference left = 1;
+  // The key to compare from the right table
+  Expression.FieldReference right = 2;
+  // Describes how to compare the two keys, defaults to the equal function
```

Comment:
When you say equals function, I assume you mean `SIMPLE_EQUALITY_TYPE_EQ`. We should specify this explicitly if so.

Reply:
I removed this line about default entirely.  We very rarely use defaults anywhere else (`buf`'s influence I think) and it's probably best not to imply that we are doing so here.